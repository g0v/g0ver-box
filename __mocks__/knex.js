import knex from 'jest-mock-knex';

export { client } from 'jest-mock-knex';

export default knex;

process.setMaxListeners(0);
