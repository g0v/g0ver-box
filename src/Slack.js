import _ from 'lodash';
import slack from 'slack';

export const bot = slack.rtm.client();

const TOKEN = process.env.SLACK_TOKEN;

if (TOKEN) {
  bot.listen({ token: TOKEN });
}

bot.hello((message) => {
  console.log(`Got a message: ${message.type}`);
});

const call = (name, req = {}) => (
  new Promise((resolve, reject) => {
    const fn = _.get(slack, name);
    fn(
      { ...req, token: TOKEN },
      (err, res) => (err ? reject(err) : resolve(res))
    );
  })
);

export default {
  userInfo: async req => await call('users.info', req),
  postMessage: async req => await call('chat.postMessage', { ...req, as_user: true }),
  postMultiMessage: async (users, message) => _.map(users, user => bot.ws.send(JSON.stringify({
    id: `${user}::${Date.now()}`, type: 'message', channel: user, text: message,
  }))),
  channelInfo: async req => await call('channels.info', req),
  channelJoin: async req => await call('channels.join', req),
};
