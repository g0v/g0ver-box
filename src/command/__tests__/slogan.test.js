import { client } from 'knex';
import Slack from '../../Slack';
import index from '../';

describe('add command', () => {
  const data = { channel: 'D100', name: 'yutin', text: 'slogan are you freestyle' };

  it('when g0ver does not exist', async () => {
    client.mockReturnValueOnce([]);
    client.mockReturnValueOnce(['yutin']);
    client.mockReturnValueOnce([]);
    await index(data);
    expect(client).toMatchSnapshot();
    expect(Slack.postMessage).toHaveBeenLastCalledWith({ channel: 'D100', text: 'done it, setting are you freestyle.' });
  });

  it('when g0ver is existed', async () => {
    client.mockReturnValueOnce([]);
    await index({ ...data, text: 'slogan 你就是沒有人' });
    expect(client).toMatchSnapshot();
    expect(Slack.postMessage).toHaveBeenLastCalledWith({ channel: 'D100', text: 'done it, setting 你就是沒有人.' });
  });
});
