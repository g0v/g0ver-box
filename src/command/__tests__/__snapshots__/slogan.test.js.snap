// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add command when g0ver does not exist 1`] = `
Array [
  "select * from g0ver where id in (U03B2AB13) and deleted_at is null",
  "insert into g0ver (created_at, id, keyword, updated_at) values (DATE, U03B2AB13, , DATE) returning id",
  "update g0ver set keyword = , slogan = are you freestyle, updated_at = DATE where deleted_at is null and id = U03B2AB13",
]
`;

exports[`add command when g0ver is existed 1`] = `
Array [
  "select * from g0ver where id in (U03B2AB13) and deleted_at is null",
  "update g0ver set keyword = , slogan = 你就是沒有人, updated_at = DATE where deleted_at is null and id = U03B2AB13",
]
`;
